eva/expression/expression_utils.py

```diff
@@ -28,3 +35,234 @@ def expression_tree_to_conjunction_list(expression_tree):
         expression_list.append(expression_tree)
 
     return expression_list
+
+
+def conjuction_list_to_expression_tree(
+    expression_list: List[AbstractExpression],
+) -> AbstractExpression:
+    """Convert expression list to expression tree wuing conjuction connector
+
+    Args:
+        expression_list (List[AbstractExpression]): list of conjunctives
+
+    Returns:
+        AbstractExpression: expression tree
+    """
+    if len(expression_list) == 0:
+        return None
+    prev_expr = expression_list[0]
+    for expr in expression_list[1:]:
+        prev_expr = LogicalExpression(ExpressionType.LOGICAL_AND, prev_expr, expr)
+    return prev_expr
+
+
+def extract_range_list_from_comparison_expr(
+    expr: ComparisonExpression, lower_bound: int, upper_bound: int
+) -> List:
+    """Extracts the valid range from the comparison expression.
+    The expression needs to be amongst <, >, <=, >=, =, !=.
+
+    Args:
+        expr (ComparisonExpression): comparison expression with two children
+            that are leaf expression nodes. If the input doesnot match,
+            the function return False
+        lower_bound (int): lower bound of the comparison predicate
+        upper_bound (int): upper bound of the comparison predicate
+
+    Returns:
+        List[Tuple(int)]: list of valid ranges
+
+    Raises:
+        RuntimeError: Invalid expression
+
+    Example:
+        extract_range_from_comparison_expr(id < 10, 0, inf): True, [(0,9)]
```

Comment:
How do we input `inf`?

Reply:
Inf is just an example. For actual code, we would always know the upper and lower bound. The ultimate goal is to get it from catalog and pass it down the pipeline.